daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/BinaryNumberUnparsers.scala

```diff
@@ -93,6 +93,27 @@ abstract class BinaryIntegerBaseUnparser(e: ElementRuntimeData, signed: Boolean)
       dos.putLong(asLong(value), nBits, finfo)
     }
   }
+
+  override def unparse(state: UState): Unit = {
+    val nBits = getBitLength(state)
```

Comment:
This is going to call getBitLength twice, once here and once when super is called. We should probably avoid that.

Also, it looks like we don't have any checks on the max size for numbers like we do in parse. Maybe we should just do all the min/max length checks in the main unparse function like we do with parse? Or maybe in BinaryIntegerBaseUnparser.putNumber?

Reply:
Hmm, the issue is putNumber doesn't have access to state, which we need for Unparse error and BinaryNumberBaseUnparser is used by non Binary Integer classes

Reply:
When there is no access to state, I think the technique is to throw a thin-exception to someplace that catches it that has the state needed to report the error in context properly.